src/java/voldemort/client/ClientConfig.java

```diff
@@ -1094,6 +1106,50 @@ public ClientConfig setMaxBootstrapRetries(int maxBootstrapRetries) {
         this.maxBootstrapRetries = maxBootstrapRetries;
         return this;
     }
+    
+    public int getBootstrapRetryWaitTimeSeconds() {
+        return this.bootstrapRetryWaitTimeSeconds;
+    }
+
+    /**
+     * set the wait time in seconds after a failed bootstrap attempt to next
+     * retry
+     * 
+     * @param value
+     *            wait time in seconds
+     * @throws IllegalArgumentException
+     *             If bootstrapWaitTimeInSeconds < 1
+     */
+    public ClientConfig setBootstrapRetryWaitTimeSeconds(int bootstrapWaitTimeInSeconds) {
+        if (bootstrapWaitTimeInSeconds < 1) {
+            throw new IllegalArgumentException("bootstrap retry wait time should be >= 1");
+        }
+        this.bootstrapRetryWaitTimeSeconds = bootstrapWaitTimeInSeconds;
+        return this;
+    }
+
+    public boolean isFetchAllStoresXmlInBootstrap() {
+        return fetchAllStoresXmlInBootstrap;
+    }
+
+    /**
+     * Voldemort servers older than 1.8.1 supported only fetching all stores.xml
+     * 1.8.1 supported fetching individual stores. During bootstrap this
+     * property controls whether to fetch all stores.xml or only the particular
+     * store.
+     * 
+     * This property is defaulted to true, so that newer clients can work with
+     * older servers. If there are 100's of stores and you have servers running
+     * on 1.8.1 or above, it is better to set them to false.
+     * 
+     * @param value
+     *            true, to fetch all stores.xml false, to fetch only current
+     *            store
+     */
+    public ClientConfig setFetchAllStoresXmlInBootstrap(boolean value) {
```

Comment:
I agree with you that it can be automatic.
1) Protocol upgrade is a tricky issue  considering Forward/Backward compatibility.
2) Admin Port should ideally be port blocked and opened only from specific boxes, given Voldemort does not have any security built into it.
3) For LinkedIn, I am planning to override the default in the wrapper to do only retrieve one store.